src/Workspaces/Core/Portable/PatternMatching/PatternMatcher.cs

```diff
@@ -52,7 +58,20 @@ internal abstract partial class PatternMatcher : IDisposable
             bool allowFuzzyMatching = false)
         {
             culture = culture ?? CultureInfo.CurrentCulture;
+            if (culture.Name == "en-US")
```

Comment:
Changing behavior like this for one specific culture is very troubling. For one thing, it won't behave the same for en-GB, en-CA, en-AU, etc. as en-US. (Mind you, VS sets current culture to match its active language pack, so you may be able to rely on en-US here based on that, but I don't know which other processes might invoke this code.)

Regardless, even when current culture is non-English, ASCII-range identifiers, file names, etc. remain exceptionally common.

Instead of using ordinal for english and culture-sensitive for other languages, I would prefer an approach that did a fast comparison attempt and fell back to the culture-sensitive comparison if any non-ASCII characters are encountered. This is the approach we took in the System.Refelection.Metadata string comparer: https://source.dot.net/#System.Reflection.Metadata/System/Reflection/Internal/Utilities/MemoryBlock.cs,378